src/bin/cargo_stable_mir_json.rs

```diff
@@ -0,0 +1,179 @@
+use std::env;
+
+use std::io::Write;
+use std::os::unix::fs::PermissionsExt;
+use std::path::{Path, PathBuf};
+
+use anyhow::{bail, Result};
+
+fn main() -> Result<()> {
+    let args: Vec<_> = env::args().collect();
+
+    let (repo_dir, maybe_user_provided_dir): (PathBuf, Option<PathBuf>) = match &args[..] {
+        [_, repo_dir] => (repo_dir.into(), None),
+        [_, repo_dir, user_provided_dir] => (repo_dir.into(), Some(user_provided_dir.into())),
+        _ => bail!("Usage: cargo run --bin cargo_stable_mir_json -- <PATH-TO-STABLE-MIR-JSON-REPO> [OPTIONAL-PATH-TO-CREATE-BUILD-DIR]"),
+    };
+
+    if !repo_dir.is_dir() {
+        bail!(
+            "Provided should be path to stable_mir_json repo, but {} is not a dir",
+            repo_dir.display()
+        );
+    }
+
+    if let Some(ref user_provided_dir) = maybe_user_provided_dir {
+        if !user_provided_dir.is_dir() {
+            bail!(
+                "Provided should be path to create the .stable_mir_json dir, but {} is not a dir",
+                user_provided_dir.display()
+            );
+        }
+    }
+
+    setup(repo_dir, maybe_user_provided_dir)
+}
+
+fn setup(repo_dir: PathBuf, maybe_user_provided_dir: Option<PathBuf>) -> Result<()> {
+    let smir_json_dir = smir_json_dir(maybe_user_provided_dir)?;
+    println!("Creating {} directory", smir_json_dir.display());
+    std::fs::create_dir(&smir_json_dir)?; // This errors is the directory already exists
+
+    let ld_library_path = record_ld_library_path(&smir_json_dir)?;
+    copy_artefacts(&repo_dir, &smir_json_dir, &ld_library_path)
+}
```

Comment:
Maybe print a "success" message here?
```suggestion
    copy_artefacts(&repo_dir, &smir_json_dir, &ld_library_path)
    println!("Artefacts installed in {}.", &simr_json_dir);
    println!("To use stable-mir-json, set `RUSTC={}/\{debug,release\}/stable-mir-json` when calling `cargo`.", smir_json_dir);
}
```